x/merkledb/db.go

```diff
@@ -1172,11 +1184,11 @@ func (db *merkleDB) getEditableNode(key path, hasValue bool) (*node, error) {
 // Editing the returned node affects the database state.
 // Returns database.ErrNotFound if the node doesn't exist.
 // Assumes [db.lock] is read locked.
-func (db *merkleDB) getNode(key path, hasValue bool) (*node, error) {
+func (db *merkleDB) getNode(key Path, hasValue bool) (*node, error) {
 	switch {
 	case db.closed:
 		return nil, database.ErrClosed
-	case key == RootPath:
+	case key == db.rootPath:
```

Comment:
Should we use `.Equals` here?

Reply:
We could, but == and .Equals are effectively equivalent.  Deleting .Equals to reduce confusion.